From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: JustMangoT <38831897+JustMangoT@users.noreply.github.com>
Date: Tue, 1 Mar 2022 20:10:45 +0700
Subject: [PATCH] Relocate server files

World containers is not set by default in bukkit.yml

diff --git a/src/main/java/gg/pufferfish/pufferfish/PufferfishConfig.java b/src/main/java/gg/pufferfish/pufferfish/PufferfishConfig.java
index 56330536c52fa327ef89d7a08e72557c6633c8bb..537641788c5c88277843fe7986f730f3ae6dd1e0 100644
--- a/src/main/java/gg/pufferfish/pufferfish/PufferfishConfig.java
+++ b/src/main/java/gg/pufferfish/pufferfish/PufferfishConfig.java
@@ -55,8 +55,8 @@ public class PufferfishConfig {
 	}
 	
 	public static void load() throws IOException {
-		File configFile = new File("pufferfish.yml");
-		
+		File configFile = new File("config", "pufferfish.yml"); // Atelier - Relocate files
+
 		if (configFile.exists()) {
 			try {
 				config.load(configFile);
diff --git a/src/main/java/net/minecraft/server/Main.java b/src/main/java/net/minecraft/server/Main.java
index 1e0d261439255091a6f61485c0747231fbd5b1db..fd50447e4a2e86a1fde6ee3eeef531972e4cea71 100644
--- a/src/main/java/net/minecraft/server/Main.java
+++ b/src/main/java/net/minecraft/server/Main.java
@@ -141,12 +141,12 @@ public class Main {
             org.spigotmc.SpigotConfig.disabledAdvancements = spigotConfiguration.getStringList("advancements.disabled"); // Paper - fix SPIGOT-5885, must be set early in init
             // Paper start - fix SPIGOT-5824
             File file;
-            File userCacheFile = new File("usercache.json");
+            File userCacheFile = new File("storage", "usercache.json"); // Atelier - Relocate files
             if (optionset.has("universe")) {
                 file = (File) optionset.valueOf("universe"); // CraftBukkit
                 userCacheFile = new File(file, "usercache.json");
             } else {
-                file = new File(bukkitConfiguration.getString("settings.world-container", "."));
+                file = new File("worlds"); // Atelier - Use 'worlds' folder by default
             }
             // Paper end - fix SPIGOT-5824
             YggdrasilAuthenticationService yggdrasilauthenticationservice = new com.destroystokyo.paper.profile.PaperAuthenticationService(Proxy.NO_PROXY); // Paper
diff --git a/src/main/java/net/minecraft/server/players/PlayerList.java b/src/main/java/net/minecraft/server/players/PlayerList.java
index 111553443a7247db92b15819e9bbd717e4ace61b..71113c8051e28d52f02f2352ee1936e483e92fd2 100644
--- a/src/main/java/net/minecraft/server/players/PlayerList.java
+++ b/src/main/java/net/minecraft/server/players/PlayerList.java
@@ -122,10 +122,12 @@ import org.bukkit.event.player.PlayerRespawnEvent;
 
 public abstract class PlayerList {
 
-    public static final File USERBANLIST_FILE = new File("banned-players.json");
-    public static final File IPBANLIST_FILE = new File("banned-ips.json");
-    public static final File OPLIST_FILE = new File("ops.json");
-    public static final File WHITELIST_FILE = new File("whitelist.json");
+    // Atelier start - Relocate files
+    public static final File USERBANLIST_FILE = new File("storage", "banned-players.json");
+    public static final File IPBANLIST_FILE = new File("storage", "banned-ips.json");
+    public static final File OPLIST_FILE = new File("storage", "ops.json");
+    public static final File WHITELIST_FILE = new File("storage", "whitelist.json");
+    // Atelier end
     private static final Logger LOGGER = LogUtils.getLogger();
     private static final int SEND_PLAYER_INFO_INTERVAL = 600;
     private static final SimpleDateFormat BAN_DATE_FORMAT = new SimpleDateFormat("yyyy-MM-dd 'at' HH:mm:ss z");
diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
index 23e2ad3be1786cfd3bf614d5fa464a70b0a73360..05b45d7aabd94f0f4b2bc8a2ec582ccfe35b5f29 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
@@ -1073,7 +1073,10 @@ public final class CraftServer implements Server {
 
     @SuppressWarnings({ "unchecked", "finally" })
     private void loadCustomPermissions() {
-        File file = new File(this.configuration.getString("settings.permissions-file"));
+        // AtelierMC start - Relocate files
+        String path = this.configuration.getString("settings.permissions-file");
+        File file = new File((path != null && path.contains("/")) ? path : "config/" + path);
+        // AtelierMC end
         FileInputStream stream;
 
         try {
diff --git a/src/main/java/org/bukkit/craftbukkit/Main.java b/src/main/java/org/bukkit/craftbukkit/Main.java
index 46251aa9de5e0131a05262818cd2c8a2b8970365..e626d55dd3bf6b44eb8a94793c0ff3fde4f8e013 100644
--- a/src/main/java/org/bukkit/craftbukkit/Main.java
+++ b/src/main/java/org/bukkit/craftbukkit/Main.java
@@ -110,13 +110,13 @@ public class Main {
                 acceptsAll(Main.asList("b", "bukkit-settings"), "File for bukkit settings")
                         .withRequiredArg()
                         .ofType(File.class)
-                        .defaultsTo(new File("bukkit.yml"))
+                        .defaultsTo(new File("config", "bukkit.yml")) // Atelier - Relocate files
                         .describedAs("Yml file");
 
                 acceptsAll(Main.asList("C", "commands-settings"), "File for command settings")
                         .withRequiredArg()
                         .ofType(File.class)
-                        .defaultsTo(new File("commands.yml"))
+                        .defaultsTo(new File("config", "commands.yml")) // Atelier - Relocate files
                         .describedAs("Yml file");
 
                 acceptsAll(Main.asList("forceUpgrade"), "Whether to force a world upgrade");
@@ -135,7 +135,7 @@ public class Main {
                 acceptsAll(Main.asList("S", "spigot-settings"), "File for spigot settings")
                         .withRequiredArg()
                         .ofType(File.class)
-                        .defaultsTo(new File("spigot.yml"))
+                        .defaultsTo(new File("config", "spigot.yml")) // Atelier - Relocate files
                         .describedAs("Yml file");
                 // Spigot End
 
@@ -143,7 +143,7 @@ public class Main {
                 acceptsAll(asList("paper", "paper-settings"), "File for paper settings")
                         .withRequiredArg()
                         .ofType(File.class)
-                        .defaultsTo(new File("paper.yml"))
+                        .defaultsTo(new File("config", "paper.yml")) // Atelier - Relocate files
                         .describedAs("Yml file");
 
                 acceptsAll(asList("add-plugin", "add-extra-plugin-jar"), "Specify paths to extra plugin jars to be loaded in addition to those in the plugins folder. This argument can be specified multiple times, once for each extra plugin jar path.")
@@ -157,7 +157,7 @@ public class Main {
                 acceptsAll(asList("purpur", "purpur-settings"), "File for purpur settings")
                     .withRequiredArg()
                     .ofType(File.class)
-                    .defaultsTo(new File("purpur.yml"))
+                    .defaultsTo(new File("config", "purpur.yml")) // Atelier - Relocate files
                     .describedAs("Yml file");
                 // Purpur end
 
@@ -298,6 +298,7 @@ public class Main {
                 // Paper end
                 System.setProperty( "library.jansi.version", "Paper" ); // Paper - set meaningless jansi version to prevent git builds from crashing on Windows
                 System.out.println("Loading libraries, please wait...");
+                java.nio.file.Files.createDirectories(java.nio.file.Paths.get("storage")); // Atelier - Relocate files
                 net.minecraft.server.Main.main(options);
             } catch (Throwable t) {
                 t.printStackTrace();
diff --git a/src/main/java/org/bukkit/craftbukkit/help/HelpYamlReader.java b/src/main/java/org/bukkit/craftbukkit/help/HelpYamlReader.java
index 74b49f4c1649c161c3c785be7708f5970989f7f3..8473ed353f243496fbb7de368c23c4bdbd69229d 100644
--- a/src/main/java/org/bukkit/craftbukkit/help/HelpYamlReader.java
+++ b/src/main/java/org/bukkit/craftbukkit/help/HelpYamlReader.java
@@ -25,7 +25,7 @@ public class HelpYamlReader {
     public HelpYamlReader(Server server) {
         this.server = server;
 
-        File helpYamlFile = new File("help.yml");
+        File helpYamlFile = new File("config", "help.yml"); // AtelierMC - Relocate files
         YamlConfiguration defaultConfig = YamlConfiguration.loadConfiguration(new InputStreamReader(getClass().getClassLoader().getResourceAsStream("configurations/help.yml"), Charsets.UTF_8));
 
         try {
